autode/utils.py

```diff
@@ -413,39 +418,40 @@ def timeout(seconds: float, return_value: Optional[Any] = None) -> Any:
         (Any): Result of the function | return_value
     """
 
-    def handler(queue, func, args, kwargs):
-        queue.put(func(*args, **kwargs))
-
     def decorator(func):
-        def wraps(*args, **kwargs):
-            q = multiprocessing.Queue()
-            p = multiprocessing.Process(
-                target=handler, args=(q, func, args, kwargs)
-            )
+        def wrapper(*args, **kwargs):
 
-            if mp.current_process().daemon:
-                # Cannot run a subprocess in a daemon process - timeout is not
-                # possible
-                return func(*args, **kwargs)
-
-            elif isinstance(mp.get_context(), mp.context.ForkContext):
-                p.start()
+            def run_func(connector):
+                pid = os.getpid()
+                connector.send(pid)  # send PID of process running job
+                result = func(*args, **kwargs)
+                return result
 
-            else:
-                logger.error("Failed to wrap function")
+            pool = loky.get_reusable_executor(max_workers=2)  # get process pool
+            conn1, conn2 = Pipe()
+            try:
+                job = pool.submit(run_func, conn1)
+            except RuntimeError:
+                logger.warn("Failed to wrap function")
                 return func(*args, **kwargs)
 
-            p.join(timeout=seconds)
-
-            if p.is_alive():
-                p.kill()
-                p.join()
-                return return_value
-
-            else:
-                return q.get()
+            job_pid = conn2.recv()  # get PID of process running job
+            try:
+                res = job.result(timeout=seconds)
+                return res
+            except loky.TimeoutError:
+                pool.shutdown(wait=False)
+                if job_pid != os.getpid():  # prevent accidentally killing main process
+                    job_proc = psutil.Process(pid=job_pid)  # attach to job process
+                    job_sub_children = job_proc.children(recursive=True)  # any subprocesses of job
+                    for proc in job_sub_children:
+                        proc.send_signal(signal.SIGTERM)  # shut-down subprocesses of job
+                    job_proc.send_signal(signal.SIGTERM)  # shut-down job process
```

Comment:
```suggestion
                    for proc in job_proc.children(recursive=True):
                        proc.send_signal(signal.SIGTERM) 
                    job_proc.send_signal(signal.SIGTERM)
```
I'm also anti temporary's unless they convey some meaning